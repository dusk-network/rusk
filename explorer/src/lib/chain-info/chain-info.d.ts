type BlockHeader = {
  date: Date;
  feespaid: number;
  hash: string;
  height: number;
  nextblockhash: string;
  prevblockhash: string;
  reward: number;
  seed: string;
  statehash: string;
};

type BlockTransactions = {
  data: Transaction[];
  stats: {
    averageGasPrice: number;
    gasLimit: number;
    gasUsed: number;
  };
};

type Block = {
  header: BlockHeader;
  transactions: BlockTransactions;
};

type ChainInfo = {
  blocks: Block[];
  transactions: Transaction[];
};

type SearchResult = {
  id: string;
  type: "account" | "block" | "transaction";
};

type Transaction = {
  amount: number | undefined;
  blobHashes: string[];
  from: string | undefined;
  to: string | undefined;
  blockhash: string;
  blockheight: number;
  date: Date;
  feepaid: number;
  gaslimit: number;
  gasprice: number;
  gasspent: number;
  memo: string;
  method: string;
  success: boolean;
  txerror: string;
  txid: string;
  txtype: string;
  payload: string;
  nonce: number | undefined;
};
