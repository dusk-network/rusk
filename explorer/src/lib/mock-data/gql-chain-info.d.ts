type GQLBlock = {
  fees: number;
  gasSpent: number;
  header: {
    gasLimit: number;
    hash: string;
    height: number;
    nextBlockHash?: string;
    prevBlockHash: string;
    seed: string;
    stateHash: string;
    timestamp: number;
  };
  reward: number;
  transactions: GQLTransaction[];
};

type GQLCallData = {
  contractId: string;
  fnName: string;
};

type GQLChainInfo = {
  blocks: GQLBlock[];
  transactions: GQLTransaction[];
};

type GQLSearchResult = {
  block?: {
    header: {
      hash: string;
    };
  } | null;
  tx?: {
    id: string;
  } | null;
  account?: {
    id: string;
  } | null;
};

type GQLTransaction = {
  blockHash: string;
  blockHeight: number;
  blockTimestamp: number;
  err: string | null;
  gasSpent: number;
  id: string;
  tx: {
    blobHashes: string[] | null;
    callData: GQLCallData | null;
    gasLimit: number;
    gasPrice: number;
    id: string;
    isDeploy: boolean;
    memo: string;
    txType: string;
    json?: string;
  };
};
