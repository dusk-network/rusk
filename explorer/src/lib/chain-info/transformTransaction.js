import { unixTsToDate } from "$lib/dusk/date";

/** @type {(v: GQLTransaction) => Transaction} */
const transformTransaction = (tx) => ({
  blockhash: tx.blockHash,
  blockheight: tx.blockHeight,
  date: unixTsToDate(tx.blockTimestamp),
  feepaid: tx.gasSpent * tx.tx.gasPrice,
  from: undefined,
  gaslimit: tx.tx.gasLimit,
  gasprice: tx.tx.gasPrice,
  gasspent: tx.gasSpent,
  memo: tx.tx.memo ?? "",
  method: tx.tx.isDeploy ? "deploy" : (tx.tx.callData?.fnName ?? "transfer"),
  success: tx.err === null,
  to: undefined,
  txerror: tx.err ?? "",
  txid: tx.id,
  txtype: tx.tx.txType,
});

export default transformTransaction;
