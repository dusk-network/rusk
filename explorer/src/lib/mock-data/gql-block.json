{
  "block": {
    "fees": 580718,
    "gasSpent": 580718,
    "header": {
      "gasLimit": 5000000000,
      "hash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
      "height": 495868,
      "nextBlockHash": "6011556208a85e6001bd01ccbf936486b91318a7f6cbcf7ab810adf6fae34204",
      "prevBlockHash": "07b74b35c2c7cf8f41426cd0870bafa1a2c7adee3fdd876643548096186fc4cb",
      "seed": "af15447e3a004a79d4ae8b084f7b76b78d95880bb63e1cfa79250a310731f52e6d84ee42a5d6fc2cb99c5b1f489761f6",
      "stateHash": "20bb0a677b93f084afadfd34bec3ac3feee33a020b81d9549afa2268e8543acb",
      "timestamp": 1713298022
    },
    "reward": 16000000000,
    "transactions": [
      {
        "blockHash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
        "blockHeight": 495868,
        "blockTimestamp": 1713298022,
        "err": null,
        "gasSpent": 290866,
        "id": "3a3f6f90a1012ae751b4448bcb8e98def0ba2b18170239bd69fcf8e2e37f0602",
        "tx": {
          "blobHashes": null,
          "callData": null,
          "isDeploy": false,
          "gasLimit": 500000000,
          "gasPrice": 1,
          "id": "3a3f6f90a1012ae751b4448bcb8e98def0ba2b18170239bd69fcf8e2e37f0602",
          "memo": "",
          "txType": "Moonlight"
        }
      },
      {
        "blockHash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
        "blockHeight": 495868,
        "blockTimestamp": 1713298022,
        "err": null,
        "gasSpent": 289852,
        "id": "07bfabea1d94c16f2dc3697fa642f6cecea6e81bf76b9644efbb6e2723b76d00",
        "tx": {
          "blobHashes": null,
          "callData": null,
          "gasLimit": 500000000,
          "gasPrice": 1,
          "id": "07bfabea1d94c16f2dc3697fa642f6cecea6e81bf76b9644efbb6e2723b76d00",
          "isDeploy": false,
          "memo": "some notes about the transaction",
          "txType": "Phoenix"
        }
      }
    ]
  }
}
