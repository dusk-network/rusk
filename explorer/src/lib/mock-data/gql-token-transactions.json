[
  {
    "amount": 4359,
    "from": "0xf1c4230aca3ca19199d0f9665bbd9cff8",
    "to": "0x189f8d571b05adc6a35ef1bb0e3b458184",
    "blockhash": "103749d1e8990cdda301b2c4efd6c50735548dd2fff511d9d99c75",
    "blockheight": 6670,
    "date": "2025-02-04 21:15:22.373891+00:00",
    "feepaid": 364,
    "gaslimit": 40133828,
    "gasprice": 21,
    "gasspent": 8255387,
    "memo": "Mock transaction",
    "method": "stake",
    "success": true,
    "txerror": "Out of gas",
    "txid": "15fb0001ed66bb0dacc02ec003de09e31a3de10e810fa8bb384175",
    "txtype": "Phoenix",
    "payload": "0x",
    "nonce": 0
  }
]
