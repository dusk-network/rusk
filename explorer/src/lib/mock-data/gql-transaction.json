{
  "tx": {
    "blockHash": "3c6e4018cfa86723e50644e33d3990bc27fc794f6b49fbf6290e4d308e07bd2d",
    "blockHeight": 487166,
    "blockTimestamp": 1713249549,
    "gasSpent": 290766,
    "id": "4877687c2dbf154248d3ddee9ba0d81e3431f39056f82a46819da041d4ac0e04",
    "err": null,
    "tx": {
      "blobHashes": null,
      "callData": null,
      "gasLimit": 500000000,
      "gasPrice": 1,
      "id": "4877687c2dbf154248d3ddee9ba0d81e3431f39056f82a46819da041d4ac0e04",
      "isDeploy": false,
      "memo": "some notes about the transaction",
      "txType": "Moonlight",
      "json": "{\"type\":\"moonlight\",\"sender\":\"kT5BamV2Xq6JpnoCq9ux7EmYRShdg4W1sLzyGFUqicmZTqxeMWLiTGpBWNhpjiPtKqvWFFmdxVBAjts3oEnZgPfuiyTRwDdASJQjTSy7wNhHdCeKtzExREcXZsmmMRLwKbXj\",\"receiver\":\"mHWZo9qYUhqp2SEmtWN7EuDKFwVrjgdEyzpHdfLB6R9scRAq5EkUQyLB9fBfCGt1wjxfEpFxHq9MBPVVPY3Lk3JKnQLWZzj7UYAR4mGmeGQZwCaeCS8uA63ZPKnGpjiUnj\",\"value\":9812378912731,\"nonce\":156,\"deposit\":0,\"fee\":{\"refund_address\":\"kT5BamV2Xq6JpnoCq9ux7EmYRShdg4W1sLzyGFUqicmZTqxeMWLiTGpBWNhpjiPtKqvWFFmdxVBAjts3oEnZgPfuiyTRwDdASJQjTSy7wNhHdCeKtzExREcXZsmmMRLwKbXj\",\"gas_limit\":\"2500000\",\"gas_price\":\"1\"},\"call\":null,\"is_deploy\":false,\"memo\":null}"
    }
  }
}
