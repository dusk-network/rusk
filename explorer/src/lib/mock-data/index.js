export { default as apiMarketData } from "./api-market-data.json";
export { default as apiNodeInfo } from "./api-node-info.json";
export { default as apiNodeLocations } from "./api-node-locations.json";
export { default as apiStats } from "./api-stats.json";
export { default as gqlBlock } from "./gql-block.json";
export { default as gqlBlockDetails } from "./gql-block-details.json";
export { default as gqlBlocks } from "./gql-blocks.json";
export { default as gqlLatestChainInfo } from "./gql-latest-chain-info.json";
export { default as gqlSearchPossibleResults } from "./gql-search-possible-results.json";
export { default as gqlTransaction } from "./gql-transaction.json";
export { default as gqlTransactions } from "./gql-transactions.json";
export { default as hostProvisioners } from "./host-provisioners.json";
export { default as nodeLocationsCount } from "./node-locations.count.json";

export { default as accounts } from "./accounts.json";
export { default as tokens } from "./tokens.json";
export { default as gqlTokenTransactions } from "./gql-token-transactions.json";
