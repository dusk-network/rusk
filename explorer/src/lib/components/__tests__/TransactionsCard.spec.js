import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { compose, mapWith, take } from "lamb";

import { gqlTransactions } from "$lib/mock-data";
import { transformTransaction } from "$lib/chain-info";
import { appStore } from "$lib/stores";

import { TransactionsCard } from "..";

describe("Transactions Card", () => {
  vi.useFakeTimers();
  vi.setSystemTime(new Date(2024, 4, 30));

  const getTenTransactions = compose(mapWith(transformTransaction), take(10));
  const data = getTenTransactions(gqlTransactions.transactions);

  const baseProps = {
    error: null,
    loading: false,
    txns: null,
    appStore: appStore
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  afterAll(() => {
    vi.useRealTimers();
  });

  it("should render the `TransactionsCard` component", () => {
    const { container } = render(TransactionsCard, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should disable the `Show More` button if the card is in the loading state", async () => {
    const loading = true;

    const { container, getByRole } = render(TransactionsCard, {
      ...baseOptions,
      props: { ...baseProps, loading },
    });

    expect(getByRole("button")).toBeDisabled();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should disable the `Show More` button if there is no more data to display", async () => {
    const loading = false;
    const txns = data;

    const { container, getByRole } = render(TransactionsCard, {
      ...baseOptions,
      props: { ...baseProps, loading, txns },
    });

    expect(getByRole("button")).toBeDisabled();

    expect(container.firstChild).toMatchSnapshot();
  });
});
