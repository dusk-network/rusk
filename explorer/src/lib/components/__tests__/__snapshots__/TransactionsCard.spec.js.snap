// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Transactions Card > should disable the \`Show More\` button if the card is in the loading state 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Transactions — 0 Displayed Items
    </h1>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--normal"
      disabled=""
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Show More
      </span>
    </button>
  </header>
   
  <div
    aria-label="Loading"
    class="dusk-progress-bar data-card__progress-bar"
    role="progressbar"
  >
    <div
      class="dusk-progress-bar__filler dusk-progress-bar__filler--undetermined"
    />
  </div>
  
   
</div>
`;

exports[`Transactions Card > should disable the \`Show More\` button if there is no more data to display 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Transactions — 10 Displayed Items
    </h1>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--normal"
      disabled=""
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Show More
      </span>
    </button>
  </header>
   
  <div
    class="data-card__content"
  >
    <div
      class="table-container transactions-table transactions-card__table"
    >
      <table
        class="table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="table__row"
          >
            <th
              class="table__header-cell"
            >
              ID
            </th>
            
             
            <th
              class="table__header-cell"
            >
              Gas
            </th>
            
             
            <th
              class="table__header-cell"
            >
              Fee (DUSK)
            </th>
            
             
            <th
              class="table__header-cell"
            >
              Status
            </th>
            
             
            <th
              class="table__header-cell"
            >
              Type
            </th>
            
          </tr>
        </thead>
         
        <tbody
          class="table__body"
        >
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=070280c69500df7d1b40942981a3b599ccf9f71ba62f96d63e322994efca2a0e"
              >
                070280c695...94efca2a0e
              </a>
               
              <small
                class="transaction__time"
              >
                18 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              50,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000215552
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=534f753d5a420fddc21c84d76ff26b87cc48ea75b450c082759d155153259c00"
              >
                534f753d5a...5153259c00
              </a>
               
              <small
                class="transaction__time"
              >
                18 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              50,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000215617
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=c6f13af20c23b8bc55d22a416a191e3a6ba47128b4d7dc3ac6fce5fc68ebb102"
              >
                c6f13af20c...fc68ebb102
              </a>
               
              <small
                class="transaction__time"
              >
                18 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              50,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000214413
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=31c7dce26a5a6ef1392f53630b8e2714981f10eaeea2685c5a37dbbf7bcb4c04"
              >
                31c7dce26a...bf7bcb4c04
              </a>
               
              <small
                class="transaction__time"
              >
                18 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              500,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000291769
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=4fa6e913c083aa595530b1f6c010c7e5a58121307c991ea6782f9e9e4dd5510d"
              >
                4fa6e913c0...9e4dd5510d
              </a>
               
              <small
                class="transaction__time"
              >
                18 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              500,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000290764
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=d1a7928a435021d87ea3649bff57c8154fc7b5c6c68081a61fea5779ccdb2e04"
              >
                d1a7928a43...79ccdb2e04
              </a>
               
              <small
                class="transaction__time"
              >
                18 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              500,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000291961
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=004ca8683c315b9270e04d3768dfb58fb0f65d92a9469ca31a5d97cff9718b06"
              >
                004ca8683c...cff9718b06
              </a>
               
              <small
                class="transaction__time"
              >
                18 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              500,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000292039
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=b5a77063e45dd90b97b2c30b4cd670fbf04ae60c4b8e3ea9fd0d0b4496253a05"
              >
                b5a77063e4...4496253a05
              </a>
               
              <small
                class="transaction__time"
              >
                19 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              500,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000290406
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=b263d9d594439be92bb9349bd97f55a05060956419f2f4a32925639d80b06e0b"
              >
                b263d9d594...9d80b06e0b
              </a>
               
              <small
                class="transaction__time"
              >
                19 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              500,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000291537
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <a
                class="dusk-anchor dusk-anchor--on-surface transaction__link"
                href="/some-base-path/transactions/transaction?id=b4525cb82f5cd56fcf8ceb2eb30f647d2c3567e82abb8265dd12daf5275d7709"
              >
                b4525cb82f...f5275d7709
              </a>
               
              <small
                class="transaction__time"
              >
                19 hours ago
              </small>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <b
                class="transaction__fee-price-label"
              >
                PRICE:
              </b>
               
              1
              <br />
               
              <b
                class="transaction__fee-limit-label"
              >
                LIMIT:
              </b>
               
              500,000,000
            </td>
            
             
            <td
              class="table__data-cell"
            >
              0.000291361
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-success"
              >
                success
              </span>
            </td>
            
             
            <td
              class="table__data-cell"
            >
              <span
                class="dusk-badge dusk-badge--variant-neutral transaction__type"
              >
                transfer
              </span>
            </td>
            
             
          </tr>
          
        </tbody>
      </table>
    </div>
    
  </div>
  
   
</div>
`;

exports[`Transactions Card > should render the \`TransactionsCard\` component 1`] = `
<div
  class="dusk-card data-card"
>
  <header
    class="data-card__header"
    slot="header"
  >
    <h1
      class="data-card__header-title"
    >
      Transactions — 0 Displayed Items
    </h1>
     
    <button
      class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--normal"
      type="button"
    >
      <span
        class="dusk-button__text"
      >
        Show More
      </span>
    </button>
  </header>
   
  
   
</div>
`;
